scripts/zynthian_envars.sh

```diff
@@ -21,28 +21,61 @@
 # For a full copy of the GNU General Public License see the LICENSE.txt file.
 # ****************************************************************************
 
-export ZYNTHIAN_KIT_VERSION="V3-PRO"
-
-#Audio Config
-export SOUNDCARD_NAME="HifiBerry DAC+ ADC PRO"
-export SOUNDCARD_CONFIG="dtoverlay=hifiberry-dacplusadcpro"
-export SOUNDCARD_MIXER="Digital,ADC,ADC Left Input,ADC Right Input"
-export JACKD_OPTIONS="-P 70 -t 2000 -s -d alsa -d hw:sndrpihifiberry -r 44100 -p 256 -n 2 -X raw"
-
-#Display Config
-export DISPLAY_NAME="ZynScreen 3.5 (v1)"
-export DISPLAY_CONFIG="dtoverlay=piscreen2r-notouch,rotate=270\ndtoverlay=ads7846,speed=2000000,cs=1,penirq=17,penirq_pull=2,swapxy=1,xohms=100,pmax=255"
-export DISPLAY_WIDTH="480"
-export DISPLAY_HEIGHT="320"
-export FRAMEBUFFER="/dev/fb1"
-
-# Zynthian Wiring Config
-export ZYNTHIAN_WIRING_LAYOUT="MCP23017_ZynScreen"
-export ZYNTHIAN_WIRING_ENCODER_A="102,105,110,113"
-export ZYNTHIAN_WIRING_ENCODER_B="101,104,109,112"
-export ZYNTHIAN_WIRING_SWITCHES="100,103,108,111,106,107,114,115"
-export ZYNTHIAN_WIRING_MCP23017_INTA_PIN="2"
-export ZYNTHIAN_WIRING_MCP23017_INTB_PIN="7"
+# Detect EPDF hat eeprom and select if present fallback to "V3-PRO" default
+if [ /zynthian/zynthian-sys/scripts/epdf_detect.sh -eq 0 ]; then
```

Comment:
This file is included on nearly every zynthian specific script/program, so i don't think it's the right place for detecting specific hardware. 
More important, this file is rewrited by webconf and zynthian-ui when some config option is modified. The changes you propose totally break the expected format and will break webconf and zynthian-ui functionality.

This kind of configuration should be done on the zynthian_update_sys.sh script that is called when needed by the webconf & zynthian-ui. I could modify the first boot code for calling it when an SD image is booted for first time, so it can correctly detect and config the hardware.

Reply:
Ahh, that makes sense that this is rewritten.  I'll take a look at zynthian_update_sys.sh and see about moving the detection and config to there. With the EPDF, since it is not a kit, I'd like it to start with the defaults on first boot, but then keep whatever user configuration is specified for Wiring, audio, and display. The Act LED and PWM fan will also be installed/configured at first boot.